packages/vitest/src/node/pools/rpc.ts

```diff
@@ -20,8 +25,31 @@ export function createMethodsRPC(project: WorkspaceProject): RuntimeRPC {
       const r = await project.vitenode.transformRequest(id)
       return r?.map as RawSourceMap | undefined
     },
-    fetch(id, transformMode) {
-      return project.vitenode.fetchModule(id, transformMode)
+    async fetch(id, transformMode) {
+      const result = await project.vitenode.fetchResult(id, transformMode)
+      const code = result.code
+      if (result.externalize)
+        return result
+      if ('id' in result)
+        return { id: result.id as string }
+
+      if (!code)
+        throw new Error(`Failed to fetch module ${id}`)
+
+      const dir = join(project.tmpDir, transformMode)
+      const tmp = join(dir, id.replace(/[/\\?%*:|"<>]/g, '_').replace('\0', '__x00__'))
+      if (promises.has(tmp)) {
+        await promises.get(tmp)
+        return { id: tmp }
+      }
+      if (!created.has(dir)) {
+        await mkdir(dir, { recursive: true })
+        created.add(dir)
+      }
+      promises.set(tmp, writeFile(tmp, code, 'utf-8').finally(() => promises.delete(tmp)))
+      await promises.get(tmp)
+      Object.assign(result, { id: tmp })
+      return { id: tmp }
```

Comment:
`result` is not used and not referenced outside the scope - is the `Object.assign` part still required?

Reply:
It's just to avoid typescript errors